third_party/move/move-bytecode-verifier/src/dependencies.rs

```diff
@@ -455,6 +455,18 @@ fn compare_types(
         (SignatureToken::Vector(ty1), SignatureToken::Vector(ty2)) => {
             compare_types(context, ty1, ty2, def_module)
         },
+        (
+            SignatureToken::Function(args1, result1, ab1),
+            SignatureToken::Function(args2, result2, ab2),
+        ) => {
+            compare_cross_module_signatures(context, args1, args2, def_module)?;
+            compare_cross_module_signatures(context, result1, result2, def_module)?;
+            if ab1 != ab2 {
+                Err(PartialVMError::new(StatusCode::TYPE_MISMATCH))
```

Comment:
Well, we can use debug and format via "{:?}"? Any extra information is better than a status code. Imagine you have debug a type mismatch - there are so many places it can come from, it is difficult to bisect, error messages allows us to pinpoint the location (if they are propagated correctly 😄)

Below is some old code I do not know who wrote, but I would love to see error messages as well there.